packages/directlinespeech/src/createAdapters.js

```diff
@@ -57,8 +60,26 @@ export default async function create({
     );
   }
 
-  if (!region || typeof region !== 'string') {
-    throw new Error('"fetchCredentials" must return "region" as a non-empty string.');
+  if ((directLineSpeechHostname && region) || (!directLineSpeechHostname && !region)) {
+    throw new Error(
+      '"fetchCredentials" must return either "directLineSpeechHostname" or "region" and it must not be empty string.'
+    );
+  }
+
+  if (directLineSpeechHostname) {
+    if (typeof directLineSpeechHostname !== 'string') {
+      throw new Error('"fetchCredentials" must return "directLineSpeechHostname" as a string.');
+    }
+
+    if (enableInternalHTTPSupport) {
+      throw new Error(
+        '"fetchCredentials" must not return "directLineSpeechHostname" if "enableInternalHTTPSupport" is set.'
```

Comment:
```suggestion
        '"fetchCredentials" must not return "directLineSpeechHostname" if "enableInternalHTTPSupport" is a set.'
```

Reply:
Skipping this one as we discussed offline.